catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/SearchFormsSymbolsIT.java

```diff
@@ -0,0 +1,438 @@
+
+      // Replace the header that the XSLT transform removed
+      return destination
+          .toString()
+          .replace(
+              "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+              "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>");
+    } catch (TransformerException e) {
+      throw new AssertionError("Could not remove pretty printing from XML, " + xml, e);
+    }
+  }
+
+  private static FilterWriter getWriter() {
+    try {
+      return new FilterWriter(true);
+    } catch (JAXBException e) {
+      throw new AssertionError("Could not make filter writer, " + e.getMessage());
+    }
+  }
+
+  private static AttributeRegistry getRegistry() {
+    AttributeRegistry registry = mock(AttributeRegistry.class);
+    doReturn(Optional.empty()).when(registry).lookup(any());
+    return registry;
+  }
+
+  private static File getResourceFile(String... resourceRoute) {
+    try {
+      File dir = new File(FILTER_RESOURCES_DIR.toURI());
+      if (!dir.exists()) {
+        fail(
+            format(
+                "Invalid setup parameter '%s', the directory does not exist",
+                FILTER_RESOURCES_DIR.toString()));
+      }
+
+      Path route = Arrays.stream(resourceRoute).map(Paths::get).reduce(Path::resolve).orElse(null);
+      if (route == null) {
+        fail("Could not reduce resource route to a single path");
+      }
+
+      File resourceFile = dir.toPath().resolve(route).toFile();
+      if (!resourceFile.exists()) {
+        fail("File was not found " + resourceFile.getAbsolutePath());
+      }
+
+      return resourceFile;
+    } catch (URISyntaxException e) {
+      throw new AssertionError("Could not complete test setup due to exception", e);
+    }
+  }
+
+  private static String getContentsOfFile(String... resourceRoute) {
+    File resourceFile = getResourceFile(resourceRoute);
+    try (FileInputStream fis = new FileInputStream(resourceFile)) {
+      return IOUtils.toString(fis, StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new AssertionError("Could not complete test setup due to exception", e);
+    }
+  }
+
+  /**
+   * Given a port range, this function returns the first port within the range that it's able to
+   * bind to. The range is inclusive on both sides.
+   *
+   * <p>The discovered port should be available for another client to bind to by the time this
+   * function has returned. Given this detail, running unit tests on environments that are in the
+   * process of being provisioned or are otherwise in a state of flux may cause erroneous failures
+   * due to port binding race conditions.
+   *
+   * @param min the lowest port to consider.
+   * @param max the highest port to consider.
+   * @return a port number the caller can <b>reasonably</b> assume is available to bind to.
+   * @throws AssertionError if the specified range was not valid or no port was available.
+   */
+  private static int getFirstAvailablePortBetween(int min, int max) {
```

Comment:
:pencil2: This could be replaced with `ServerSocket s = new ServerSocket(0); s.getLocalPort();` as it will handle finding an open port on its own. https://docs.oracle.com/javase/8/docs/api/java/net/ServerSocket.html#ServerSocket-int-